modules/em/src/DensityMap.cpp

```diff
@@ -1104,6 +1104,84 @@ DensityMap *DensityMap::get_cropped(const algebra::BoundingBox3D &bb) {
   return cropped_dmap.release();
 }
 
+DensityMap *DensityMap::get_cropped(Particles ps, float distance, bool inverse) {
+  
+  algebra::Vector3Ds coords;
+  algebra::Vector3D vox_cent;
+  float vox_dist; 
+  bool crop;
+  Pointer<DensityMap> cropped_map;
+
+  // Always good to have a list of coordinates
+  for (unsigned int i=0; i<ps.size(); i++){
+    coords.push_back(core::XYZ(ps[i]).get_coordinates());
+  }
+  
+  if(inverse==true) {
+      // For an inverse map, we simply make a clone of the 
+      // current map, since the bounds will be the same
+      cropped_map = create_density_map(this);
+  } else {
+      // If we want a resultant map contining the particles
+      // we create a new map using a bounding box based on 
+      // the particle coordinates and distance threshold
+      algebra::Vector3D mincorner(100000,100000,100000);
+      algebra::Vector3D maxcorner(-100000,-100000,-100000);
+      
+      // Get max/min coordinates of the inputted particle set
+      for(unsigned int i=0;i<coords.size();i++){
+        if(coords[i][0] < mincorner[0]) { mincorner[0]=coords[i][0];}
+        if(coords[i][0] > maxcorner[0]) { maxcorner[0]=coords[i][0];}
+        if(coords[i][1] < mincorner[1]) { mincorner[1]=coords[i][1];}
+        if(coords[i][1] > maxcorner[1]) { maxcorner[1]=coords[i][1];}
+        if(coords[i][2] < mincorner[2]) { mincorner[2]=coords[i][2];}
+        if(coords[i][2] > maxcorner[2]) { maxcorner[2]=coords[i][2];}
+      }
+
+      // build a bounding box and add the distance threshold
+      algebra::BoundingBox3D bb(mincorner, maxcorner); 
```

Comment:
There's already a constructor that takes a list of coordinates: https://integrativemodeling.org/2.14.0/doc/ref/classIMP_1_1algebra_1_1BoundingBoxD.html#a612cedf52eea6d53f4ce2d5e219541b7

If that doesn't work for you for some reason, you should add a comment explaining why here (or fix the constructor).

Reply:
Yes, since I found that it is easy to bad bounding box dimensions, I can just use that constructor.